packages/gatsby/src/query/graphql-runner.ts

```diff
@@ -170,6 +177,13 @@ export class GraphQLRunner {
       return { errors }
     }
 
+    if (warnings.length > 0) {
+      warnings.forEach(err => {
+        const message = context.path ? `\nQueried in ${context.path}` : ``
+        reporter.warn(err.message + message)
+      })
+    }
+
```

Comment:
Should this happen before early return if there are any errors? Just so warnings are not hidden until errors are fixed (I find it bit frustrating to fix errors only to discover new things popping up that I didn't know about before).

Reply:
I was hesitant about this but yeah, probably makes sense 👍